addons/website_mail/static/src/js/website_mail.js

```diff
@@ -0,0 +1,193 @@
+odoo.define('website_mail.thread', function(require) {
+    'use strict';
+
+    var web_editor_base = require('web_editor.base');
+    var ajax = require('web.ajax');
+    var core = require('web.core');
+    var Widget = require('web.Widget');
+
+    var qweb = core.qweb;
+    var _t = core._t;
+
+    /**
+     * Widget WebsiteMailThread
+     *
+     *  - Fetch message fron controller
+     *  - Display chatter: pager, total message, composer (according to access right)
+     *  - Provider API to filter displayed messages
+     */
+    var WebsiteMailThread = Widget.extend({
```

Comment:
General note: some space issues (missing spaces, ...), a linting is probably necessary.